controllers/mirroring/mirroring_controller.go

```diff
@@ -211,13 +210,16 @@ func (r *MirroringReconciler) reconcilePhases(clientMappingConfig *corev1.Config
 		return ctrl.Result{RequeueAfter: 3 * time.Second}, nil
 	}
 
+	errorOccurred := false
+
 	ocsClient, err := providerClient.NewProviderClient(r.ctx, storageClusterPeer.Spec.ApiEndpoint, util.OcsClientTimeout)
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to create a new provider client: %v", err)
+		r.log.Error(err, "failed to create a new provider client")
+		errorOccurred = true
+	}
+	if ocsClient != nil {
```

Comment:
is there a possibility that err is nil and still ocsClient isn't created or else no need for this check.

Reply:
Since I am not returning the error in the previous if block, we need to check if ocsClient is nil or not before closing it

Reply:
done